pkg/collector/corechecks/networkpath/networkpath.go

```diff
@@ -0,0 +1,134 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package networkpath defines the agent corecheck for
+// the Network Path integration
+package networkpath
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/networkpath/traceroute"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// CheckName defines the name of the
+// Network Path check
+const CheckName = "networkpath"
+
+// Check doesn't need additional fields
+type Check struct {
+	core.CheckBase
+	config        *CheckConfig
+	lastCheckTime time.Time
+}
+
+// Program constants and default values
+const (
+	DefaultSourcePort   = 12345
+	DefaultDestPort     = 33434
+	DefaultNumPaths     = 10
+	DefaultMinTTL       = 1
+	DefaultMaxTTL       = 30
+	DefaultDelay        = 50 //msec
+	DefaultReadTimeout  = 3 * time.Second
+	DefaultOutputFormat = "json"
+)
+
+// Run executes the check
+func (c *Check) Run() error {
+	startTime := time.Now()
+	senderInstance, err := c.GetSender()
+	if err != nil {
+		return err
+	}
+
+	cfg := traceroute.Config{
+		DestHostname: c.config.DestHostname,
+	}
+
+	trcrt := traceroute.New(cfg)
+	path, err := trcrt.Run()
+	if err != nil {
+		return fmt.Errorf("failed to trace path: %w", err)
+	}
+
+	// send to EP
+	err = c.SendNetPathMDToEP(senderInstance, path)
+	if err != nil {
+		return fmt.Errorf("failed to send network path metadata: %w", err)
+	}
+
+	tags := []string{
+		"dest_hostname:" + c.config.DestHostname,
+		"dest_name:" + c.config.DestName,
+	}
+	duration := time.Since(startTime)
+	senderInstance.Gauge("networkpath.telemetry.count", 1, "", tags)
+	senderInstance.Gauge("networkpath.telemetry.duration", duration.Seconds(), "", tags)
+
+	if !c.lastCheckTime.IsZero() {
+		interval := startTime.Sub(c.lastCheckTime)
+		senderInstance.Gauge("networkpath.telemetry.interval", interval.Seconds(), "", tags)
+	}
+	senderInstance.Commit()
+
+	numWorkers := config.Datadog.GetInt("check_runners")
+	senderInstance.Gauge("networkpath.telemetry.check_runners", float64(numWorkers), "", tags)
+	senderInstance.Gauge("networkpath.telemetry.hop_count", float64(len(path.Hops)), "", tags)
```

Comment:
Do you mind if we remove the telemetry metrics for the moment, I added them previously for debugging purpose, and I'm not sure yet if we need them.

I think we can add them later if we think we need them.

For example, check_runners are already reported by the agent, so, we don't need to report them here too.
And few metrics are currently not well name :)

Reply:
Yeah I think that's fine. In the end we may want them, but as we're developing we can always add metrics in our own builds